### PR TITLE
Refactor to upstream

### PR DESCRIPTION
**Pull request by Gerrit91**
- Adapts to oscommon generator from Gardener repository, makes the project ready for Gardener v1.35
  - For provisioning we return our ignition userdata as before
  - The cloud-init.sh template is now templated through Gardener mechanisms and therefore supports bootstrap tokens
- Causes a restart of the CRI engine because the cloud-init.sh template has changed
- Removes unnecessary parts from the cloud-init.sh template

### CODE REVIEW

File: pkg/generator/ignition/ignition.go [hunk 1]
```diff
@@ -32,41 +30,28 @@ ExecStart=/usr/bin/containerd --config=/etc/containerd/config.toml
 // Starting with ignition 2.0, ignition itself contains the required parsing logic, so we can use ignition directly.
 // see https://github.com/coreos/ignition/blob/master/config/config.go#L38
 // Therefore we must update ignition to 2.0.0 in the images and transform the gardener config to the ignition config types instead.
-func IgnitionFromOperatingSystemConfig(ctx context.Context, c client.Client, config *extensionsv1alpha1.OperatingSystemConfig) ([]byte, error) {
+func IgnitionFromOperatingSystemConfig(config *generator.OperatingSystemConfig) ([]byte, error) {
```
Comment (majst01) [inline pkg/generator/ignition/ignition.go:33]: We can move this into generator.go an make it private, ignition_test.go can then be renamed to genertor_test.go. The whole package is named ignition this make all obvious anyway.

Comment (Gerrit91) [reply pkg/generator/ignition/ignition.go]: Done.

**Review by majst01 — APPROVED** [review]
Maybe we should check if the containerd/dockerd config can be passed via the shoot spec

**Comment by Gerrit91** [comment]
Created an issue for this https://github.com/metal-stack/os-metal-extension/issues/22.

**Pull request merged**